setup.cfg

```diff
@@ -0,0 +1,5 @@
+[flake8]
+# this config is mostly there to help with the use of pre-commit
+max-line-length = 110
+ignore = E203,E402,E722,W503,F841
+docstring-convention = numpy
```

Comment:
I think for this to work we would need https://pypi.org/project/flake8-docstrings/ ... correct?

We could add it to `requirements.txt` and then also add it to the pre-commit hook like here: https://github.com/sappelhoff/pyprep/blob/25b91e098d0055f14b5cbcae54b5ee9e0739aa62/.pre-commit-config.yaml#L27

Reply:
Hum...

I have been "wasting" some time because the linting rules of precommit and those of the ci the bidsschematools clashed, so more than happy to try to make them agree.

However I know hate this PR because it does too. Many. Things.

So I will remove all changes that are not related to the title of this PR and put them in a different one.

Reply:
It's also fine with me to merge this as is and then iterate, no worries.

Reply:
`setup.cfg` is really meant for Python packages, which this isn't. If we're going to have it at the root, I would rather call it `.flake8`. See https://flake8.pycqa.org/en/latest/user/configuration.html